src/Controls/src/Core/CheckBox/CheckBox.cs

```diff
@@ -16,11 +18,40 @@ public partial class CheckBox : View, IElementConfiguration<CheckBox>, IBorderEl
 			BindableProperty.Create(nameof(IsChecked), typeof(bool), typeof(CheckBox), false,
 				propertyChanged: (bindable, oldValue, newValue) =>
 				{
-					((CheckBox)bindable).Handler?.UpdateValue(nameof(ICheckBox.Foreground));
-					((CheckBox)bindable).CheckedChanged?.Invoke(bindable, new CheckedChangedEventArgs((bool)newValue));
-					((CheckBox)bindable).ChangeVisualState();
+					if (bindable is not CheckBox checkBox)
+					{
+						return;
+					}
+
+					checkBox.Handler?.UpdateValue(nameof(ICheckBox.Foreground));
+					checkBox.CheckedChanged?.Invoke(bindable, new CheckedChangedEventArgs((bool)newValue));
+					if (checkBox.Command is not null && checkBox.Command.CanExecute(null))
+					{
+						checkBox.Command.Execute(checkBox.CommandParameter);
+					}
+
+					checkBox.ChangeVisualState();
 				}, defaultBindingMode: BindingMode.TwoWay);
 
+
+		public static readonly BindableProperty CommandProperty = BindableProperty.Create(nameof(Command), typeof(ICommand), typeof(CheckBox), null, propertyChanging: CommandElement.OnCommandChanging, propertyChanged: CommandElement.OnCommandChanged);
```

Comment:
The new properties must be included in the PublicAPI files by platform. You can do it directly from Visual Studio but let me know if can help.

```
D:\a\_work\1\s\src\Controls\src\Core\CheckBox\CheckBox.cs(41,4): error RS0016: Symbol 'Command.get' is not part of the declared public API (https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md) [D:\a\_work\1\s\src\Controls\src\Core\Controls.Core.csproj::TargetFramework=net9.0-android]
D:\a\_work\1\s\src\Controls\src\Core\CheckBox\CheckBox.cs(42,4): error RS0016: Symbol 'Command.set' is not part of the declared public API (https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md) [D:\a\_work\1\s\src\Controls\src\Core\Controls.Core.csproj::TargetFramework=net9.0-android]
D:\a\_work\1\s\src\Controls\src\Core\CheckBox\CheckBox.cs(49,4): error RS0016: Symbol 'CommandParameter.get' is not part of the declared public API (https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md) [D:\a\_work\1\s\src\Controls\src\Core\Controls.Core.csproj::TargetFramework=net9.0-android]
D:\a\_work\1\s\src\Controls\src\Core\CheckBox\CheckBox.cs(50,4): error RS0016: Symbol 'CommandParameter.set' is not part of the declared public API (https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md) [D:\a\_work\1\s\src\Controls\src\Core\Controls.Core.csproj::TargetFramework=net9.0-android]
D:\a\_work\1\s\src\Controls\src\Core\CheckBox\CheckBox.cs(115,4): error RS0016: Symbol 'IsEnabledCore.get' is not part of the declared public API (https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md) [D:\a\_work\1\s\src\Controls\src\Core\Controls.Core.csproj::TargetFramework=net9.0-android]
    56 Error(s)
```

Reply:
Can we add inline docs also?